geomesa-cassandra/geomesa-cassandra-datastore/src/main/scala/org/locationtech/geomesa/cassandra/data/CassandraIndexAdapter.scala

```diff
@@ -43,7 +43,8 @@ class CassandraIndexAdapter(ds: CassandraDataStore) extends IndexAdapter[Cassand
 
     if (cluster.getMetadata.getKeyspace(ds.session.getLoggedKeyspace).getTable(table) == null) {
       val columns = CassandraColumnMapper(index).columns
-      val (partitions, pks) = columns.partition(_.partition)
+      require(columns.last.name == SimpleFeatureColumnName, s"Expected final column to be ${SimpleFeatureColumnName}")
+      val (partitions, pks) = columns.dropRight(1).partition(_.partition) // drop serialized feature col
```

Comment:
Is there a test which should be updated?

Also, does this present an upgrade issue?

Reply:
we don't have a unit test for this... was that a subtle suggestion to add one? ;)
I don't think it will cause any upgrade issues, since the table columns are still the same, and it would only affect newly created tables.
I'm not a C* expert by any means - the issue was raised by a user on gitter and i was hoping to get some feedback from them, but gitter appears to be down atm.

Reply:
@andrijz are you on github?

Reply:
Oh!  This is the code which calculates the table / column names?  Makes sense!

Reply:
@elahrvivaz , yes I'm on github: https://github.com/andrijz
I'm actually wondering if you could a patch for this soon. We are working on MVP solution which is going to be released by the end of the March, and this impacts part of the functionality, i.e.: working with duplicates

Reply:
@andrijz if you are ok with this fix I can back-port it to our next 2.4 bug fix release. I'm not sure when we'll be actually cutting that though - in the meantime, you'd probably have to build your own artifacts from source.

Reply:
@elahrvivaz , I would really appreciate you would back-port it to the next 2.4 bug fix release. I'll try to build the artefacts from the source as well. 
Thank you!

Reply:
@andrijz Thanks for testing!  Let us know here, Gitter, or on the JIRA ticket https://geomesa.atlassian.net/browse/GEOMESA-2782 if you hit issues.  Good luck with your MVP!